### ISSUE TITLE
Option to filter `debug-files`

### ISSUE THREAD

**Issue by farfromrefug** [getsentry/sentry-fastlane-plugin#277]
I use the cli through the fastlane plugin for iOS [https://github.com/getsentry/sentry-fastlane-plugin](<https://github.com/getsentry/sentry-fastlane-plugin>).<br>when i do `sentry_debug_files_upload`:

```ruby
sentry_debug_files_upload(
      url: ENV["SENTRY_URL"],
      org_slug: ENV["SENTRY_ORG"],
      project_slug: ENV["SENTRY_PROJECT"],
      wait: true,
      derived_data: false,
      include_sources: false,
      no_unwind: true,
      no_sources: true,
      upload_symbol_maps: true
    )
```

It detects a lot of debug files, some i want but also a lot i dont want for an iOS project on Macos, like `.node` files, files like `ffi_c.bundle`, `libswiftos.dylib`, `sentry-cli-Darwin-universal`, `sentry-cli-Linux-x86_64`, ....

It adds a lot of upload time, and server storage for nothing.<br>A simple solution would be a filter parameter. The ideal thing would be for it to support an "array" of name or glob

**Comment by szokeasaurusrex** [comment]
@philipphofmann that is correct, what I wanted to check first in the Fastlane plugin is whether perhaps we should be passing a more specific path to Sentry CLI. As in, maybe the files @farfromrefug is referring to are in a different directory which we should not be searching for debug files. If this is the case, then the requested filtering feature may not even be needed. Since the plugin passes the path to Sentry CLI, we could change this in the plugin. 

If these files are in the same path as the debug symbols, then filtering is needed, and in that case, I agree that we should first implement the filtering in Sentry CLI.

**Comment by kahest** [comment]
@farfromrefug can you provide more information on how you use the fastlane-plugin, which folders do you provide?

**Comment by farfromrefug** [comment]
@kahest indeed i do it with Nativescript. Here is a sample project https://github.com/Akylas/conty
Please let me know how i can help. Having first class fastlane support is really important for us

**Comment by kahest** [comment]
@farfromrefug thanks for the sample, that's helpful 🙏 I can't give an ETA, but if it's a smallish change, we might be able to squeeze it in

**Comment by szokeasaurusrex** [comment]
@farfromrefug Have you tried manually passing the path to your debug files by using [the `path` argument](https://github.com/getsentry/sentry-fastlane-plugin?tab=readme-ov-file#uploading-debug-information-files) for `sentry_debug_files_upload`?

Providing the path explicitly might solve your problem

**Comment by farfromrefug** [comment]
> [@farfromrefug](https://github.com/farfromrefug) Have you tried manually passing the path to your debug files by using [the `path` argument](https://github.com/getsentry/sentry-fastlane-plugin?tab=readme-ov-file#uploading-debug-information-files) for `sentry_debug_files_upload`?
> 
> Providing the path explicitly might solve your problem, since the default is to search the entire project directory, which is likely too broad (I have raised this issue separately in [#290](https://github.com/getsentry/sentry-fastlane-plugin/issues/290)).

I havent tried. Will try when i can, however i already know it wont be sufficient. Might need to upload files from different directories so i have to use a "parent" directory which might include unwanted files (i dont have control over the file pattern).
I think a filter (glob) would help solve that

**Comment by philprime** [comment]
I'm working on the next major release of this plugin. What is the status on this issue?

**Comment by philippsuess** [comment]
@farfromrefug, is this issue still relevant for you? Did you solve your use case differently? If yes, how?

**Comment by farfromrefug** [comment]
@philipphofmann yes this is still relevant I did not find a fix for this. would be awesome to be able to filter

**Comment by philprime** [comment]
As the fastlane plugin is a wrapper around sentry-cli, this functionality would need to be implemented in sentry-cli first, then exposed through the plugin.

## Use Case

When using `sentry_debug_files_upload` with a parent directory that contains multiple subdirectories, unwanted files are being picked up. For example, in a NativeScript project structure:

```
dist/
├── platforms/
│   └── ios/
│       └── build/
│           └── MyApp.app.dSYM/          # ✅ Want this
├── node_modules/
│   └── native-lib/
│       └── ffi_c.bundle                 # ❌ Don't want this
│       └── libswiftos.dylib            # ❌ Don't want this
└── bin/
    └── sentry-cli-Darwin-universal     # ❌ Don't want this
```

If we pass `path: 'dist'` to search for dSYM files across multiple iOS build outputs, the scanner also picks up `.node` files, `.bundle` files, `sentry-cli` binaries, and other non-debug files from `node_modules` and other directories.

## Proposed Solution

I propose adding two options to sentry-cli's `debug-files upload` command:

- **`--include <glob>`**: A glob pattern (e.g., `**/*.dSYM`, `**/*.dSYM/**`) to include only files whose full paths match the pattern. This option should also prevent traversal into directory hierarchies that don't match the pattern (for performance).

- **`--exclude <glob>`**: A glob pattern (e.g., `**/node_modules/**`, `**/*.node`) to exclude directories or files from the search if they match the pattern. Can be specified multiple times or use comma-separation (whatever is easier to do).

## Why Both Options?

Both `--include` and `--exclude` provide flexibility:

1. **`--include`** is useful when you know the exact pattern of files you want (e.g., `**/*.dSYM/**` for iOS projects).
2. **`--exclude`** is useful when you want to filter out known unwanted directories/files (e.g., `**/node_modules/**`, `**/*.node`, `**/sentry-cli-*`).

They can also be used together: include a broad pattern and exclude specific unwanted paths.

## Alternative Consideration

A single `--filter` option could work, but having both `--include` and `--exclude` is more explicit and aligns with common CLI patterns (e.g., `rsync`, `tar`, `git`).

@szokeasaurusrex Would implementing both `--include` and `--exclude` in sentry-cli be reasonable? Once available in sentry-cli, we can expose them through the fastlane plugin.

@farfromrefug would this solve your issue?

**Comment by farfromrefug** [comment]
@philprime that look awesome and yes include and exclude seems best

**Comment by philprime** [comment]
@szokeasaurusrex do you have bandwidth to implement this in the CLI or do you want me to create a PR?

**Comment by szokeasaurusrex** [comment]
@philprime not at the moment. Feel free to do a PR. I suggest following the pattern for the source maps uploading and injecting commands. We already support something similar there
